scripts/tcl_commands/synthesis.tcl

```diff
@@ -243,29 +243,68 @@ proc logic_equiv_check {args} {
     exec echo "[TIMER::get_runtime]" | python3 $::env(SCRIPTS_DIR)/write_runtime.py "logic equivalence check - yosys"
 }
 
+
+proc generate_blackbox_verilog {inputs output {defines ""}} {
+    set output_files ""
+    set defines_flag ""
+    set ::env(YOSYS_IN) $inputs
+    set ::env(YOSYS_OUT) $output
+    if { $defines != "" } {
+        set ::env(YOSYS_DEFINES) $defines
+    }
+    try_exec yosys -c $::env(SCRIPTS_DIR)/yosys/blackbox.tcl
+    puts_info "Generated blackbox verilog ($output) from ($inputs)"
+}
+
+
 proc run_verilator {} {
-    set verilator_verified_pdks "sky130A sky130B"
-    set verilator_verified_scl "sky130_fd_sc_hd"
-    set includes ""
-    if { [string match *$::env(PDK)* $verilator_verified_pdks] == 0 || \
-        [string match *$::env(STD_CELL_LIBRARY)* $verilator_verified_scl] == 0} {
-        puts_warn "PDK '$::env(PDK)', SCL '$::env(STD_CELL_LIBRARY)' will generate errors with instantiated stdcells in the design."
-        puts_warn "Either disable QUIT_ON_LINTER_ERRORS or remove the instantiated cells."
-    } else {
-        set pdk_verilog_models [glob $::env(PDK_ROOT)/$::env(PDK)/libs.ref/$::env(STD_CELL_LIBRARY_OPT)/verilog/*.v]
-        foreach model $pdk_verilog_models {
-            set includes "$includes $model"
-        }
+    set pdk_model_blackbox ""
+    if { $::env(PDK) == "sky130A" ||$::env(PDK) == "sky130B" } {
+        set pdk_model "$::env(PDK_ROOT)/$::env(PDK)/libs.ref/$::env(STD_CELL_LIBRARY)/verilog/$::env(STD_CELL_LIBRARY)__blackbox.v"
+        set output_file "$::env(synthesis_tmpfiles)/[file rootname [file tail $pdk_model]]-bb.v"
+        generate_blackbox_verilog $pdk_model $output_file
+
+        exec echo "\n/* verilator lint_off UNDRIVEN */\n" | cat - $output_file > $output_file.tmp
+        file rename -force $output_file.tmp $output_file
+        exec echo "\n/* verilator lint_off UNUSEDSIGNAL */\n" | cat - $output_file > $output_file.tmp
+        file rename -force $output_file.tmp $output_file
+        exec echo "\n/* verilator lint_on UNUSEDSIGNAL */\n" >> $output_file
+        exec echo "\n/* verilator lint_on UNDRIVEN */\n" >> $output_file
+
+        set pdk_model_blackbox "$pdk_model_blackbox $output_file"
+    }
+    if { ($::env(PDK) == "gf180mcuC" || $::env(PDK) == "gf180mcuA" || $::env(PDK) == "gf180mcuB") && \
+        ($::env(STD_CELL_LIBRARY) == "gf180mcu_fd_sc_mcu7t5v0" || $::env(STD_CELL_LIBRARY) == "gf180mcu_fd_sc_mcu9t5v0")} {
+        set pdk_model_original "$::env(PDK_ROOT)/$::env(PDK)/libs.ref/$::env(STD_CELL_LIBRARY)/verilog/$::env(STD_CELL_LIBRARY).v"
+        set pdk_model_patched "$::env(synthesis_tmpfiles)/[file rootname [file tail $pdk_model_original]]-patched.v"
```

Comment:
THIS part on the other hand should be an if statement for `gf180mcu{A,B,C}`- the two if-statements are largely identical